elastica/rod/factory_function.py

```diff
@@ -216,7 +215,43 @@ def allocate(
         )
 
     # Shear/Stretch matrix
-    shear_modulus = youngs_modulus / (poisson_ratio + 1.0)
+    if kwargs.__contains__("shear_modulus"):
+        shear_modulus = kwargs.get("shear_modulus")
+        if kwargs.__contains__("poisson_ratio"):
+            poisson_ratio = kwargs.get("poisson_ratio")
+            message = (
+                " Poisson ratio ( "
```

Comment:
I would suggest:
"Both a Poisson ratio and a shear modulus are provided. The Poisson ratio is only used to compute a shear modulus so the provided Poisson ratio of ( "+ str(poisson_ratio)+ " ) is being ignored in favor of the provided shear modulus ( "+ str(shear_modulus)+ " ) \n"